pkg/resources/deployments/deployment.go

```diff
@@ -125,8 +129,8 @@ func mcsContainer(mi *miniov1beta1.MinIOInstance) corev1.Container {
 	}
 }
 
-// NewForCluster creates a new Deployment for the given MinIO instance.
-func NewForCluster(mi *miniov1beta1.MinIOInstance) *appsv1.Deployment {
+// NewMcsDeployment creates a new Deployment for the given MinIO instance.
+func NewMcsDeployment(mi *miniov1beta1.MinIOInstance) *appsv1.Deployment {
```

Comment:
Shouldn't we be redefining the `MCS_MINIO_SERVER` to point the new mcs service endpoint here in `mcsEnvironmentVars` function.

Reply:
the MCS_MINIO_SERVER already uses `mi.GetServiceHost()` which returns the new load balanced service